tools/branch_comparisons.md

```diff
@@ -0,0 +1,107 @@
+# Branch comparisons
+
+These links take to comparisons between branches.
+
+If a cell in the table is `-`, this means that the two Ignition releases use
+the same version of this Ignition library. So, for example, `ign-cmake` has a
+`-` for B ➡️  C because Blueprint and Citadel use `ign-cmake2`.
+
+When it says "There isn’t anything to compare," this means no pull requests
+need to be forward-ported.
+
+Otherwise, those branches could use a forward-port pull request, see
+instructions on the
+[contributing guide](https://ignitionrobotics.org/docs/all/contributing#process).
+
+* **B**: Blueprint
+* **C**: Citadel
+* **D**: Dome
+* **main**: main branch
+
+> SDFormat is a special case where the release branches should be compared
+  against `merge_N_M_reference` branches instead of other release branches.
+  Be sure to update those branches before using for comparison.
```

Comment:
```suggestion
> SDFormat is a special case that doesn't follow the forward-port process listed in the contributing guide.
  For SDFormat, the release branches should be compared against `merge_N_M_reference` branches (where `M = N + 1`) instead of other release branches.
  Be sure to update the `merge_N_M_reference` branches before using them for comparison.
  The `merge_N_M_reference` branches need to be updated if `compare/merge_N_M_reference...sdfN` isn't empty.
  If this comparison isn't empty, update `merge_N_M_reference` by merging `sdfN` into `merge_N_M_reference`.
  After updating `merge_N_M_reference`, see what's in `merge_N_M_reference` that hasn't been merged forward to `sdfM` yet by looking at the **diff** of `compare/sdfM...merge_N_M_reference` (the list of commits in this comparison isn't helpful), and cherry-pick accordingly.
```

I tried to clarify this a little more based on this [comment](https://github.com/ignitionrobotics/docs/pull/108#discussion_r509484526), but I'm not 100% sure I understand how the cherry-picking part works - do we cherry-pick commits into a separate branch that is then used to create a forward-port PR?

Reply:
SDFormat is using the same pattern as the rest now, so this isn't needed.